cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -359,7 +356,12 @@ private BodyConsumer deployApplication(final HttpResponder responder,
     }
 
     if (archiveName == null || archiveName.isEmpty()) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, ARCHIVE_NAME_HEADER + " header not present",
+      responder.sendString(HttpResponseStatus.BAD_REQUEST,
+                           String.format(
+                             "%s header not present. Please include the header and set its value to the jar name. " +
```

Comment:
nope there is no api for that.  You can include a configuration json in the old api, but only as a header, which is odd.